engine/src/test/java/org/hibernate/search/test/configuration/sharding/ShardingConfigurationTest.java

```diff
@@ -262,6 +266,21 @@ private MutableSearchFactory getSearchFactory(Map<String, String> shardingProper
 		).buildSearchIntegrator();
 	}
 
+	private static Path getTargetDir() {
+		URI classesDirUri;
+		try {
+			classesDirUri = ShardingConfigurationTest.class.getProtectionDomain()
+					.getCodeSource()
+					.getLocation()
```

Comment:
Yes, it must not be invoked for classes from the classpath. Hence I moved that into modules (see my introductory comment at this PR) and changed it to not accept a class object which suggests more flexibility than actually present.

Regarding the implementation, I'd say it's much simpler. No functional benefit apart from that, just my "canonical way" of doing this :)